src/artemisSocket.js

```diff
@@ -21,7 +17,14 @@ class Socket extends net.Socket {
 
 		this._debug = !!options.debug;
 		this._buffer = null;
-		this.on('data',this._parseData);
+		this.on('data', this._parseData);
+
+		this.on('error', this._onError);
```

Comment:
I don't think this is needed - the functionality of `net.Socket` can take care of panicking on critical errors, I think.